src/main/java/org/project_kessel/relations/client/RelationsGrpcClientsManager.java

```diff
@@ -27,6 +42,21 @@ public static synchronized RelationsGrpcClientsManager forSecureClients(String t
         return secureManagers.get(targetUrl);
     }
 
+    public static synchronized RelationsGrpcClientsManager forSecureClients(String targetUrl, Config.AuthenticationConfig authnConfig) {
+        if (!secureManagers.containsKey(targetUrl)) {
```

Comment:
If we can add additional implementation to pass token, for example: ```forSecureClientWithToken(String target, Config.AuthenticationConfig authnConfig, jwtToken string)```

If the client application already manages the token generation part, it can pass the token string to RelationsGrpcClientsManager. In such case it will reduce the token generation request to sso?

Reply:
I think in that case, I would be inclined to create a new AuthnMode and new block in `Config` called something like "OIDC_CLIENT_CREDENTIALS_SUPPLIED" where the token would then be passed in as part of `Config.AuthenticationConfig authnConfig`. What do you think?

I think it's probably a follow-up jira to be honest.

Reply:
yes, followup jira would be best for that use-case